### DISCUSSION TITLE
Regarding migration of odoo versions using OpenUpgrade.

### DISCUSSION THREAD

**Discussion by SateeshMathapati** [OCA/OpenUpgrade#4541] [Q&A]
Dear All, 
I have two questions regarding migration scripts of odoo.

1. When can we expect OpenUpgrade 17.0 migration scripts?
2. Can we modify existing OpenUpgrade 16.0 to make it work for Odoo 17.0 version? If yes, How to modify scripts ?

Please help me on this.
Thank you.

**Comment by vdewulf** [comment]
Hello,

There is no timeline for the OpenUpgrade 17.0 migration scripts yet. You can followup the progress here: 
https://github.com/OCA/OpenUpgrade/issues/4124

A small group of active OpenUpgrade contributors will soon be working more actively on those scripts, so we can expect a part of them to get ready (for the first main Applications) in the next months.

For your second question, if only it would be so easy! That is why the OpenUpgrade scripts take time to get ready. It requires a lot of work to get the scripts correctly written in each new version, depending on the data model changes by Odoo in the standard Odoo code.
You can learn more about the process of creating the scripts in the OpenUpgrade official documentation: 
https://oca.github.io/OpenUpgrade/020_required_knowledge.html#to-develop-openupgrade-scripts

Have a nice day!
